app/views/handlers/list_add_question.py

```diff
@@ -9,13 +9,29 @@ def is_location_valid(self):
             return False
         return True
 
+    def get_next_location_url(self):
+        return self.parent_location.url()
```

Comment:
This will need to persist the query params. A test will need to be added to ensure this is working. You will most likely need to add another block after the list collector to be able to test this.

```py
        return self.parent_location.url(
            return_to=self._return_to,
            return_to_answer_id=self._return_to_answer_id,
            return_to_block_id=self._return_to_block_id,
        )
```

Reply:
Query params added